### ISSUE TITLE
Import folders as playlists

### ISSUE THREAD

**Issue by p-i-c-o** [OxygenCobalt/Auxio#811]
### Description

I want to be able to import folders that contain mp3s that are then saved as playlists

### Other implementations

No other players really do this but this would help a lot because i have many folders with different playlists from spotify that I exported using spotdl

### Benefit

This would help me save time be importing playlists instead of re-creating them manually

### Duplicates

- [X] I have checked this feature request for duplicates.
- [X] I have checked that this feature is not implemented in the [lastest version](https://github.com/OxygenCobalt/Auxio/releases).
- [X] I have checked the [Why Are These Features Missing?](https://github.com/OxygenCobalt/Auxio/wiki/Why-Are-These-Features-Missing%3F) page.
- [X] I agree to the [Contribution Guidelines](https://github.com/OxygenCobalt/Auxio/blob/dev/.github/CONTRIBUTING.md).

**Comment by OxygenCobalt** [comment]
Was planning to do this at *some* point depending on user demand. Will keep this as a maybe right now.

**Comment by j3bx** [comment]
please add this enhancement

**Comment by p-i-c-o** [comment]
> Was planning to do this at _some_ point depending on user demand. Will keep this as a maybe right now.

This is literally the only reason I don't use Auxio, please consider.

**Comment by ConEstArthur** [comment]
Do consider please.

**Comment by EKliptus** [comment]
I need this too,
I  can't use Auxio. I need the ability to  browse by folder or add a folder as playlist too...

**Comment by OxygenCobalt** [comment]
Approving this from signifigant demand, but implementation is still unclear.

**Comment by landry314** [comment]
yes! also, a folder view tab because all my music is already sorted into folders on my phone.

**Comment by OxygenCobalt** [comment]
> yes! also, a folder view tab because all my music is already sorted into folders on my phone.

This is not planned, Auxio is not a folder based music player since folder libraries differ significantly from tagged libraries @landry314

**Comment by OxygenCobalt** [comment]
My idea is that a folder-based playlist would work something like #932. That would allow it to be meaningfully abstracted enough while also making it remain synced w/the device long-term.

**Comment by OxygenCobalt** [comment]
I actually have a question: How many people want this because they legitimately put songs into playlist-like folders, and how many people want this as an informal replacement for a folder view?

There seems to be a lot of demand on the order of wanting frankly just a folder view. If the latter, I'm not really keen on accepting this. It will just be a even worse version of a folder view, and I still do not want to add folder views when autotagging is trivial now.

**Comment by OxygenCobalt** [comment]
Flipping back to maybe until I can see a legitimately significant demand for this that isn't just because they want to use a folder based library.

**Comment by landry314** [comment]
to create a subset of the audio files on the phone by chosing only certain folders to scan?

**Comment by OxygenCobalt** [comment]
No, effectively if they have a folder of untagged music for some artists, they want to use this feature to make a playlist for that artist, functionally making a folder-based library. If this is the features primary use, I don't want to add this since it's functionally a bad version of a folder view.

**Comment by gonssal** [comment]
I have all my music classified by folders in the following way: `Genre / Artist / Song.mp3`. So for example `Rock / Elvis Presley / Burning Love.mp3`. I've been looking for a music player where I can browse my SD Card, select either the Genre or Artist folders and have an option like "Add folder to playlist", so I can easily create a "Rock" playlist, or an "Elvis Presley" playlist. To me it's mindblowing that I still haven't found an Android music player making this possible, and sadly Auxio can't either, so I'm going to try the next one.

**Comment by Glockko** [comment]
I just started using the app and everything is perfect but this feature is the only thing that's missing for me. I have my music in folders consisting of songs from many different albums and artists. All I would like is to be able to play songs from a folder as if from a playlist. If this isn't possible in some shape or form I will have to keep looking for another app because I'm not going to be adding 1000+ songs manually into playlists. Mr Developer thank you for this app but I seriously hope you reconsider.

**Comment by Glockko** [comment]
I can already browse my folders outside of the app, if only I could play songs by folder (or import a folder as a playlist) that would be amazing.

**Comment by OxygenCobalt** [comment]
> Many singles and instrumental tracks do not have artist or album information. If they cannot be displayed based on the file name and the folder directory structure, it would be extremely chaotic. Therefore, I believe that at the very least, the folder directory tree should be available as one of the sorting criteria.

> I can already browse my folders outside of the app, if only I could play songs by folder (or import a folder as a playlist) that would be amazing.

But this is the issue. I cannot meaningfully parse the metadata information of every single user's library from the folder structure alone, and making a "generalized" folder UI will result in extreme jank and instability. This is why I avoid filesystem views and recommend the use of autotaggers like beets or MusicBrainz Picard (seriously, I imagine those tools can ID the metadata for your tracks in most cases unless it was not released publically at all) @time-of-flowers @Glockko 

> I understand the reasoning behing using metadata instead as an organization tool. The problem with that is that metadata isn't always correct, and modifying it is a nightmare, virtually unfeasible on big music libraries. Also this is not asking for Auxio to organize music based on folders structure, it's just asking allowing all the songs in a folder to be added to a playlist, also allowing to create a new one if desired instead.

Autotaggers are really powerful in this regard, I'd recommend trying them out @gonssal

**Comment by Andrew15-5** [comment]
I randomly looked at a new list of apps in Accrescent, found this app, thought I'd finally have a free app with no bs, and I can't use it, because it doesn't have a directory based navigation. Since genres are not always the same as playlists, it's impossible for me to use it or create playlists from thousands of entries in the default list.

I have all my songs saved to different directories, which makes it super easy to directly listen to a directory of songs, or create a playlist from that. Which is easy to do in Clementine, Samsung Music, Stellio Player. But the last two are awful for various reasons, and there is no Clementine for Android.

So I'm stuck with a subpar solution for times when I can't use online music service. I mean, I can technically use Auxio, but without my actual playlist, with this many tracks, it's just unusable. I don't listen to all songs, only to custom playlists.

So, I'm gonna wait for this, hopefully finding salvation after.

P.S. I mean, the app looks very promising and cool, apart from the re-loading of all songs hiccup when opening it after closing.

**Comment by WhatIfBlueberry** [comment]
I would also really appreciate this feature. I have multiple playlists as folders on my desktop pc and moved them to my phone. I tried creating .m3u files for each folder, which worked fine for VLC (the app) but when importing them in Auxio the created Playlists contain only a random number of songs (like 30 of 500, differs for each Playlist). Currently i see no way of recreating those playlists except manually selecting thousands of songs. 

Love the App! Please keep up the great work.

**Comment by asuhadayo** [comment]
Development has stalled on my previous preferred music player Vanilla Music, and I was looking for an alternative. This looked good... Except for the fact that I can't browse the filesystem structure... Creating playlists by scanning folders could probably sidestep this issue, but most of my songs aren't tagged... And auto-tagging won't work, as over half of the music I have has no public release (outside M3 or Comiket, at least)... I have about 12,000 songs, organized by file-system structure, somewhat inconsistently into genre and then artist, which makes it much easier for me to queue up several artists from a particular genre of my choosing... I don't know how much jank it'd cause, but at the bare minimum, some form of this feature would be necessary for me to make use of this effectively.

If there's no plan to implement this feature, then it's probably not going to be the app for me. Either way, thank you for you hard work, clearly there's a lot of people that very much so appreciate it. I just hope some time in the future I may be able to as well.

**Comment by zencharcoal** [comment]
I would also like to see this feature. M3U files are manageable and scriptable on a laptop, but this will break once in the app since file paths won't match OR it will duplicate playlist tracks and eat up storage space. Hmm, maybe if M3U file is local to that folder? Investigating and I will report back. 

If my music library was perfectly organized, it wouldn't be a problem, but after so many years and various sources, it gets cumbersome to manage. Generally, appreciating the app though. 

Mostly just hoping to make playlist creation easier and simpler in general, since that seems to be the major struggle we all have with out libraries.

**Comment by zencharcoal** [comment]
> I would also like to see this feature. M3U files are manageable and scriptable on a laptop, but this will break once in the app since file paths won't match OR it will duplicate playlist tracks and eat up storage space. Hmm, maybe if M3U file is local to that folder? Investigating and I will report back.
> 

Made a quick script effort, but limited success. One of two problems. M3U file must contain full paths OR import function must work with relative paths.  If you can share the full file path expected by the import function, I will happily provide you and whomever with my bash script.
